libraries/AP_Math/matrixN.cpp

```diff
@@ -0,0 +1,216 @@
+/*
+ *  Library for operations on matrices of arbitrary dimensions.
+ *
+ *  Based on the MatrixMath library created by Charlie Matlack on 12/18/10.
+ *  Modified from code by RobH45345 on Arduino Forums, taken from unknown source.
+ */
+
+#pragma GCC optimize("O3")
+ 
+#include "matrixN.h"
+
+void matrix_copy(float* A, int32_t n, int32_t m, float* B){
+    int32_t i, j;
```

Comment:
Because I was porting this code and didn't give it much thought :) You are right, replaced with uint8_t.